docs/reference/api/usb/USBAudio.md

```diff
@@ -2,22 +2,59 @@
 
 <span class="images">![](https://os.mbed.com/docs/v5.9/feature-hal-spec-usb-device-doxy/class_u_s_b_audio.png)<span>USBAudio class hierarchy</span></span>
 
-You can use the USBAudio interface to send and receive audio across USB. By selecting a USBAudio device as your PC's speaker output, you can send audio data. You can also use it as your PC's microphone and receive sound sent by USBAudio.
+You can use the USBAudio interface to send and receive audio data over USB. Once a USB program is loaded onto the Mbed board, you can send audio data to your PC by selecting **Mbed Audio** as your PC's microphone input, and you can have the Mbed board receive audio data from your PC by selecting **Mbed Audio** as your PC's speaker output.
 
 ### USBAudio class reference
 
 [![View code](https://www.mbed.com/embed/?type=library)](https://os.mbed.com/docs/v5.9/feature-hal-spec-usb-device-doxy/class_u_s_b_audio.html)
 
-### USBAudio example
+### USBAudio square wave example
 
 ```C++
+// This example generates a square wave. 
+// Use a recording program, such as Audacity, to record and hear the square wave, or route microphone input to output device.
 #include "mbed.h"
 #include "USBAudio.h"
+#include <math.h>
 
+int16_t square_wave(uint32_t freq_hz, uint16_t amplitude, float time_s)
+{
+    float period = (float)1 / freq_hz;
+    if (fmod(time_s, period) > period / 2) {
+        return amplitude / 2;
+    } else {
+        return -(amplitude / 2);
+    }
+}
+
+int main() {
+    uint32_t tx_freq = 16000;
+    USBAudio audio(true, 8000, 2, tx_freq, 1, 10, 0x7bb8, 0x1112, 0x0100);
+    float cur_time = 0;
+    while (true) {
+        uint16_t samples[64];
+        for (int i = 0; i < 64; i++) {
+            samples[i] = square_wave(100, 5000, cur_time);
+            cur_time += 1.0 / tx_freq;
+        }
+        if (!audio.write((uint8_t *)&samples, sizeof(samples))) {
+            audio.write_wait_ready();
+        }
+    }
+}
+
+```
+
+### USBAudio loopback example
```

Comment:
Same comment as above for a small introduction to set the scene here as well for the loopback example.